lib/state_machines/integrations/active_record.rb

```diff
@@ -441,8 +441,12 @@ def runs_validations_on_action?
       # Gets the db default for the machine's attribute
       def owner_class_attribute_default
         if owner_class.connected? && owner_class.table_exists?
-          if column = owner_class.columns_hash[attribute.to_s]
-            column.default
+          if ::ActiveRecord.gem_version >= Gem::Version.new('4.2.0')
```

Comment:
Instead of adding the conditional inside the method I think it is better to conditionally define the method. Can you change that?